tools/themis/src/rules.rs

```diff
@@ -0,0 +1,329 @@
+
+    if !outliers.is_empty() {
+        return Err(Error::OutcomeError {
+            msg: "These non-private packages should have a `license` specification".to_string(),
+            expected: None,
+            outliers,
+        });
+    }
+
+    Ok(())
+}
+
+/// ensure all non-private crates have a license file
+///
+/// Checks for either one LICENSE file, or two LICENSE files, one of which
+/// is the Apache License 2.0 and the other is the MIT license.
+pub fn publishable_has_license_file(workspace: &Workspace) -> Result<(), Error> {
+    let outliers = workspace
+        .members
+        .iter()
+        .filter(|pkg| {
+            utils::is_publishable(pkg)
+                && !utils::exists!(
+                    pkg,
+                    "LICENSE" || ("LICENSE-APACHE" && "LICENSE-MIT") || (pkg.parsed.license_file)?
+                )
+        })
+        .map(|pkg| PackageOutcome {
+            pkg,
+            value: pkg.parsed.license_file.as_ref().map(|l| l.to_string()),
+        })
+        .collect::<Vec<_>>();
+
+    if !outliers.is_empty() {
+        return Err(Error::OutcomeError {
+            msg: "These non-private packages should have a license file".to_string(),
+            expected: None,
+            outliers,
+        });
+    }
+
+    Ok(())
+}
+
+const EXPECTED_LICENSE: &str = "MIT OR Apache-2.0";
+
+/// ensure all non-private crates use the the same expected license
+pub fn publishable_has_unified_license(workspace: &Workspace) -> Result<(), Error> {
+    let outliers = workspace
+        .members
+        .iter()
+        .filter(|pkg| {
+            utils::is_publishable(pkg)
+                && pkg.parsed.license.as_ref().map_or(false, |l| l != EXPECTED_LICENSE)
+        })
+        .map(|pkg| PackageOutcome {
+            pkg,
+            value: pkg.parsed.license.as_ref().map(|l| l.to_string()),
+        })
+        .collect::<Vec<_>>();
+
+    if !outliers.is_empty() {
+        return Err(Error::OutcomeError {
+            msg: "These non-private packages have an unexpected license".to_string(),
+            expected: Some(Expected { value: EXPECTED_LICENSE.to_string(), reason: None }),
+            outliers,
+        });
+    }
+
+    Ok(())
+}
+
+/// ensure all non-private crates have a description
+pub fn publishable_has_description(workspace: &Workspace) -> Result<(), Error> {
+    let outliers = workspace
+        .members
+        .iter()
+        .filter(|pkg| utils::is_publishable(pkg) && pkg.parsed.description.is_none())
+        .map(|pkg| PackageOutcome { pkg, value: None })
+        .collect::<Vec<_>>();
+
+    if !outliers.is_empty() {
+        return Err(Error::OutcomeError {
+            msg: "These non-private packages should have a `description`".to_string(),
+            expected: None,
+            outliers,
+        });
+    }
+
+    Ok(())
+}
+
+/// ensure all non-private crates have a README file
+pub fn publishable_has_readme(workspace: &Workspace) -> Result<(), Error> {
```

Comment:
This too, but maybe when we actually have all crates documented. For now, we can prioritize non-private crates.